src/installer.ts

```diff
@@ -93,10 +92,48 @@ export function getPlatform() : string | undefined {
 	const isMac = (os === 'darwin');
 	const isLinux = (os === 'linux');
 
-    if (isWindows) { return PLATFORM_WINDOWS; }
-    if (isMac) { return PLATFORM_MAC; }
-    if (isLinux) { return PLATFORM_LINUX; }
-    return undefined;
+	if (isWindows) { return PLATFORM_WINDOWS; }
+	if (isMac) { return PLATFORM_MAC; }
+	if (isLinux) { return PLATFORM_LINUX; }
+	return undefined;
+}
+
+function downloadAndExtract(link : string, dlFilename: string, installFolder : string, extract : boolean) : Promise<boolean> {
+	return new Promise<boolean>( async (resolve, reject) => {
+		const downloadSettings = {
+			filename: `${dlFilename}`,
+			extract: extract,
+		};
+		extension.mainOutputChannel.appendLine('Downloading from: ' + link);
+		let progressTitle : string = `Download (${dlFilename})`;
+		vscode.window.withProgress({
+			location: vscode.ProgressLocation.Notification,
+			title: progressTitle,
+			cancellable: false
+		}, async (progress, token) => {
+			progress.report({ increment: 0 });
+			let lastProg:number = 0;
+			let success:boolean = false;
+			await download(link, installFolder, downloadSettings)
+				.on('response', (response) => {
+					extension.mainOutputChannel.appendLine(`Bytes to transfer: ${response.headers['content-length']}`);
+				})
+				.on('downloadProgress', (progressamt) => {
+					let incr = progressamt.total > 0 ? Math.floor((progressamt.transferred - lastProg) / progressamt.total * 100) : 0;
+					if (incr >= 1) {
+						progress.report({increment: incr, message: `${Math.floor(progressamt.transferred / progressamt.total * 100)}%`});
+						lastProg = progressamt.transferred;
+					}
+					if (progressamt.transferred >= progressamt.total) {
+						success = true;
+					}
+				});
+			resolve(success);
+		})
+		.then( () => {
+			extension.mainOutputChannel.appendLine('done!');
```

Comment:
as the channel is used to log different things, i think we should specify what is done.